rpmbuild/bin/copr-builder-ready

```diff
@@ -0,0 +1,100 @@
+#! /usr/bin/python3
+
+"""
+Final checks that the builder machine is ready to be used
+
+Everything printed to STDOUT will be redirected to the copr-backend logs,
+STDERR will be ignored.
+"""
+
+import os
+import sys
+import time
+from fnmatch import fnmatch
+from subprocess import run, PIPE
+from copr_rpmbuild.config import Config
+
+
+def check_mock_config(chroot):
+    """
+    Does the mock config for this chroot exist?
+    """
+    if chroot == "srpm-builds":
+        return
+
+    config = "/etc/mock/{}.cfg".format(chroot)
+    if os.path.isfile(config):
+        return
+
+    print("Chroot config {} not found".format(config))
+    sys.exit(1)
+
+
+def subscription_required(chroot):
+    """
+    Is subscription required for this task?
+    """
+    config = Config()
+    config.load_config()
+
+    for pattern in config.rhsm:
+        if fnmatch(chroot, pattern):
+            return True
+    return False
+
+
+def active_subscription():
+    """
+    Is subscription active on this system?
+    """
+    # This implementation requires root privileges which we fortunately have
+    # when calling this script. In case this function needs to be re-written to
+    # be used under a normal user, it could be done by checking the existence of
+    # the `/etc/pki/consumer/cert.pem` file. However, it will break in the
+    # following corner cases:
+    #   - The system is registered and then halted for a long time and then
+    #     booted again after the entitlement is no longer valid
+    #   - The system is unregistered from the server (but not from the client
+    #     itself), making all the client files stale
+    cmd = ["subscription-manager", "status"]
```

Comment:
so we don't need the `subscription-manager attach`, right?

Reply:
Okay, it's stupid and you were correct all along.

I disabled the subscription-manager ansible tasks, so I can go through them manually.

```
$ subscription-manager status
$ echo $?
1
```

After registering, status already returns status zero.

```
$ subscription-manager register --force --username copr-team --password ... --name unknown-builder
$ subscription-manager status
$ echo $?
0
```

So did the cert file recommended by the subscription-manager devs

```
$ ls /etc/pki/consumer/cert.pem
/etc/pki/consumer/cert.pem
```

For the record, the certficates you use in Mock doesn't exist yet

```
ls /etc/pki/entitlement/*.pem
ls: cannot access '/etc/pki/entitlement/*.pem': No such file or directory
```

After successfully attaching, they appear

```
$ subscription-manager attach --pool 8a85f9a17c71102f017ce611251c770f
$ ls -1 /etc/pki/entitlement/*.pem |wc -l
6
```

But when my first attempt to attach failed

```
$ subscription-manager attach --pool 8a85f9a17c71102f017ce611251c770f
Successfully attached a subscription for: Red Hat Enterprise Linux for Open Source Infrastructure
Network error, unable to connect to server. Please see /var/log/rhsm/rhsm.log for more information.
```

I was too fast to re-run the command and didn't check if those entitlements appeared or not. But on re-running, this message showed

```
$ subscription-manager attach --pool 8a85f9a17c71102f017ce611251c770f
Successfully attached a subscription for: Red Hat Enterprise Linux for Open Source Infrastructure
1 local certificate has been deleted.
```

so it is possible something was created even if we didn't attach successfully. I can try throughout the day, if I can hit the issue again but looks like this is leading to us creating `/var/run/copr-subscribed` file once our `try_indefinitely subscription-manager attach` finishes.

Reply:
Thank you for this complex experiment!  Do you think it's sufficient to
submit a new RFE to the RHSM team so they understand our use case?  In the
current "cloud" era, our scenario doesn't seem particularly unusual.  It
would be awesome to have these things "RHSM built-in".